src/text/TextComponent.cpp

```diff
@@ -0,0 +1,261 @@
+#include <format>
+#include <iostream>
+
+#include <scene/components/MeshComponent.h>
+#include <scene/components/TransformComponent.h>
+#include <text/TextComponent.h>
+
+namespace arch::text {
+
+TextComponent::TextComponent(ecs::Domain& domain, const ecs::Entity entity) noexcept:
+	_domain{ &domain },
+	_entity{ entity } {
+	_domain
+		->addComponent<scene::components::TransformComponent>(_entity, float3{}, Quat{ 0, 0, 0, 1 }, float3{ 1, 1, 0 });
```

Comment:
This shouldn't be here.
It's users responsibility to add all required components, not component's them selfs.